src/output.rs

```diff
@@ -265,6 +271,54 @@ impl OutputTypeDetection for TxOut {
         }
         false
     }
+    /// Checks if an output is a OP_RETURN output meeting the requirements
+    /// for a resuable payment code
+    ///
+    /// A payment code notification transaction contains an OP_RETURN output
+    /// with 80-byte payload. The script pubkey is of the structure
+    /// OP_RETURN(0x6a) OP_PUSHDATA1(0x4c) 80-bytes(0x50)
+    ///
+    fn is_opreturn_bip47_payment_code(&self) -> bool {
+        let script_pubkey_bytes = self.script_pubkey.as_bytes();
+        // Check the length and structure
+        if script_pubkey_bytes.len() != 83
+            && script_pubkey_bytes[0] != 0x6A
+            && script_pubkey_bytes[1] != 0x4C
+            && script_pubkey_bytes[2] != 0x50
+        {
+            return false;
+        }
+        // Examine the payload
+        let payload = &script_pubkey_bytes[3..];
+        // Byte 0 - version should be 0x01 or 0x02
+        if payload[0] != 0x01 && payload[0] != 0x02 {
+            return false;
+        }
+        // Byte 2 - Sign should be 0x02 or 0x03
+        let sign_byte = payload[2];
+        if sign_byte != 0x02 && sign_byte != 0x03 {
+            return false;
+        }
+        // Bytes 3-34 - x value, must be a member of the secp256k1 group
+        let x_coordinate = &payload[3..35];
+        let mut pubkey_bytes = Vec::with_capacity(33);
+        pubkey_bytes.push(sign_byte);
+        pubkey_bytes.extend_from_slice(x_coordinate);
+        if secp256k1::PublicKey::from_slice(&pubkey_bytes).is_err() {
+            return false;
+        }
```

Comment:
As shown with the testcases added in d4434a306ef8821db7b2918d32b97e8c5b96a588, there seems to be a bug here. For some of the (valid) pubkeys, it returns InvalidPubkeyError in `secp256k1::PublicKey::from_slice()`. I haven't had time to debug this closer, but my guess is that the `pubkey_bytes` you are constructing doesn't work. Probably, some of the pubkeys work as they encode the x-value and others don't work as they encode the y-value?

Reply:
Had another look at this. We can't check if the x-value is a pubkey, because it's masked and we can't unmask it because we aren't the receiver of the notification transaction. Checking that it's not all zeros should be enough for our use case.